### PR TITLE
Optimize the C++ code after running code quality check (Complete)

### PR DESCRIPTION
**Pull request by jeevithiesh**

### CODE REVIEW

File: goa/Classes/ext/dragonBones/animation/Animation.cpp [hunk 1]
```diff
@@ -468,12 +468,9 @@ bool Animation::isCompleted() const
             return false;
         }
 
-        for (const auto animationState : _animationStates)
+        if(std::any_of(_animationStates.begin(), _animationStates.end(), [](auto animateState){return !animateState->isCompleted()}))
```
Comment (ankitk20) [inline goa/Classes/ext/dragonBones/animation/Animation.cpp]: Format this if condition correctly

File: goa/Classes/ext/dragonBones/armature/Slot.h [hunk 2]
```diff
@@ -82,14 +82,11 @@ class Slot : public TransformObject
 
     inline bool _isMeshBonesUpdate() const
     {
-        for (auto bone : _meshBones)
+        if(std::any_of(_meshBones.begin(), _meshBones.end(), [](auto bone){return (bone->_transformDirty != Bone::BoneTransformDirty::None);}))
```
Comment (ankitk20) [inline goa/Classes/ext/dragonBones/armature/Slot.h]: Format this if condition correctly

File: goa/Classes/ext/lipiengine/LipiEngineModule.cpp [hunk 3]
```diff
@@ -219,7 +219,7 @@ int LTKLipiEngineModule::createShapeRecognizer(const string& strProjName,
 		"Entering: LTKLipiEngineModule::createShapeRecognizer()"<<endl;
 	
 	int errorCode;
-    int iResult = 0;
+    //int iResult;
```
Comment (ankitk20) [inline goa/Classes/ext/lipiengine/LipiEngineModule.cpp]: Remove if not required

File: goa/Classes/ext/reco/shaperec/featureextractor/pointfloat/PointFloatShapeFeatureExtractor.cpp [hunk 4]
```diff
@@ -168,7 +168,7 @@ int PointFloatShapeFeatureExtractor::extractFeatures(const LTKTraceGroup& inTrac
         "PointFloatShapeFeatureExtractor::extractFeatures()" << endl;
     
     PointFloatShapeFeature *featurePtr = NULL;
-	float x,y,deltax;
+	//float x,y,deltax;
```
Comment (ankitk20) [inline goa/Classes/ext/reco/shaperec/featureextractor/pointfloat/PointFloatShapeFeatureExtractor.cpp]: Remove if not required

File: goa/Classes/ext/util/lib/LTKCheckSumGenerate.cpp [hunk 5]
```diff
@@ -109,7 +109,7 @@ void LTKCheckSumGenerate::initCRC32Table()
 	{
 		m_CRC32Table[i]=reflect(i, 8) << 24;
 		for (int j = 0; j < 8; j++)
-			m_CRC32Table[i] = (m_CRC32Table[i] << 1) ^ (m_CRC32Table[i] & (1 << 31) ? ulPolynomial : 0);
+			m_CRC32Table[i] = (m_CRC32Table[i] << 1) ^ ((m_CRC32Table[i] & (-2147483648)) ? ulPolynomial : 0);
```
Comment (ankitk20) [inline goa/Classes/ext/util/lib/LTKCheckSumGenerate.cpp]: Remove brackets around the integers

File: goa/Classes/sqlite3/sqlite3.c [hunk 6]
```diff
@@ -18824,7 +18824,9 @@ SQLITE_PRIVATE void *sqlite3Malloc(int n){
   }else{
     p = sqlite3GlobalConfig.m.xMalloc(n);
   }
-  assert( EIGHT_BYTE_ALIGNMENT(p) );  /* IMP: R-04675-44850 */
+  if(p != NULL){
+      assert( EIGHT_BYTE_ALIGNMENT(p) );
```
Comment (ankitk20) [inline goa/Classes/sqlite3/sqlite3.c:18828]: Format this if correctly

File: goa/Classes/sqlite3/sqlite3.c [hunk 7]
```diff
@@ -86404,7 +86394,7 @@ static void randomFunc(
     ** 2s complement of that positive value.  The end result can
     ** therefore be no less than -9223372036854775807.
     */
-    r = -(r ^ (((sqlite3_int64)1)<<63));
+    r = -(r ^ (-9223372036854775807));//1 << 63 is -9,223,372,036,854,775,807
```
Comment (ankitk20) [inline goa/Classes/sqlite3/sqlite3.c]: Parenthesis not required around integer

File: goa/cocos2d/external/recast/DetourTileCache/DetourTileCacheBuilder.cpp [hunk 8]
```diff
@@ -1083,7 +1083,7 @@ inline int area2(const unsigned char* a, const unsigned char* b, const unsigned
 //	(This idea is due to Michael Baldwin.)
 inline bool xorb(bool x, bool y)
 {
-	return !x ^ !y;
+	return (bool)(!x ^ !y);
```
Comment (ankitk20) [inline goa/cocos2d/external/recast/DetourTileCache/DetourTileCacheBuilder.cpp]: Add a space after _(bool)_

File: goa/proj.android-studio/downloader_library/src/main/res/values-v9/styles.xml [hunk 9]
```diff
@@ -1,5 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+
     <style name="NotificationText" parent="android:TextAppearance.StatusBar.EventContent" />
+
```
Comment (ankitk20) [inline goa/proj.android-studio/downloader_library/src/main/res/values-v9/styles.xml]: Remove blank lines

File: goa/proj.android-studio/goa/src/main/java/chimple/DownloadExpansionFile.java [hunk 10]
```diff
@@ -377,14 +378,14 @@ private void setState(int newState) {
...
-     * Go through each of the Expansion APK files and open each as a zip file.
-     * Calculate the CRC for each file and return false if any fail to match.
-     *
-     * @return true if XAPKZipFile is successful
-     */
+//    /**
+//     * Go through each of the Expansion APK files and open each as a zip file.
+//     * Calculate the CRC for each file and return false if any fail to match.
+//     *
+//     * @return true if XAPKZipFile is successful
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/java/chimple/DownloadExpansionFile.java]: Remove single line comment

File: goa/proj.android-studio/goa/src/main/java/com/maq/xprize/bali/repo/LessonRepo.java [hunk 11]
```diff
@@ -287,11 +294,11 @@ private static void convertToUniqueSubjects(List<FlashCard> lucs, boolean caseIn
                 FlashCard luc = lucs[lucIndex];
                 String answer = luc.objectUnit.name;
                 String[] answers = wordMap.get(answer);
-                Set<String> choiceCloneSet = new HashSet<String>(choiceSet);
-                for (int a = 0; a < answers.length; a++) {
-                    choiceCloneSet.remove(answers[a]);
+                Set<String> choiceCloneSet = new HashSet<>(choiceSet);
+                for (String answer1 : answers) {
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/java/com/maq/xprize/bali/repo/LessonRepo.java]: Give a better name. Discuss with me if required.

File: goa/proj.android-studio/goa/src/main/java/com/maq/xprize/bali/service/ServiceRunnable.java [hunk 12]
```diff
@@ -17,17 +17,17 @@ public void run() {
...
-    /**
-     * Called when an exception is thrown
-     *
-     * @param e
-     */
+//    /**
+//     * Called when an exception is thrown
+//     *
+//     * @param e
+//     */
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/java/com/maq/xprize/bali/service/ServiceRunnable.java]: Remove single line comment

File: goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/SplashScreenActivity.java [hunk 13]
```diff
@@ -117,7 +115,7 @@ public void unzipFile() {
             }
             toCallApplication();
         } catch (IOException e) {
-            System.out.println(e);
+            System.out.println("An Exception has occured");
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/SplashScreenActivity.java]: Make e small

File: goa/proj.android-studio/goa/src/main/res/xml/backup_descriptor.xml [hunk 14]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<full-backup-content>
```
Comment (ankitk20) [inline goa/proj.android-studio/goa/src/main/res/xml/backup_descriptor.xml]: Remove backup_descriptor.xml for now

File: goa/proj.android-studio/libcocos2dx/src/main/java/org/cocos2dx/lib/Cocos2dxLocalStorage.java [hunk 15]
```diff
@@ -37,14 +37,16 @@ of this software and associated documentation files (the "Software"), to deal
...
+
     private static DBOpenHelper mDatabaseOpenHelper = null;
     private static SQLiteDatabase mDatabase = null;
-    /**
-     * Constructor
-     * @param context The Context within which to work, used to create the DB
-     * @return 
-     */
+
+//    /**
```
Comment (ankitk20) [inline goa/proj.android-studio/libcocos2dx/src/main/java/org/cocos2dx/lib/Cocos2dxLocalStorage.java]: Remove single line comment

**Review by ankitk20 — CHANGES_REQUESTED** [review]

**Review by ankitk20 — DISMISSED** [review]

File: goa/Classes/AppDelegate.cpp [hunk 16]
```diff
@@ -195,13 +195,9 @@ int AppDelegate::printExtendedSceneGraph(int fd, Node *node, int level)
...
     //        Console::Utility::sendToConsole(fd, "-", 1);
-    float x = 0.0;
-    float y = 0.0;
     if (node->getParent() != nullptr)
     {
         auto nodeInWorld = node->getParent()->convertToWorldSpace(node->getPosition());
-        x = nodeInWorld.x;
-        y = nodeInWorld.y;
     }
     //    Console::Utility::mydprintf(fd, " %s x=%f y=%f\n", node->getDescription().c_str(), x, y);
```
Comment (mathewnik90) [inline goa/Classes/AppDelegate.cpp:203]: Remove it

File: goa/Classes/ext/LipiTKNode.cpp [hunk 17]
```diff
@@ -127,14 +127,11 @@ bool LipiTKNode::initialize(int width, int height, Point position) {
     return true;
 }
 
-
```
Comment (mathewnik90) [inline goa/Classes/ext/LipiTKNode.cpp:130]: Fix formatting

File: goa/Classes/ext/LipiTKNode.cpp [hunk 18]
```diff
@@ -297,9 +294,9 @@ void LipiTKNode::removeClearButton() {
 	_drawingBoard->removeChild(_clearButton, true);
 }
 
-void LipiTKNode::setClearButtonTexture(const std::string normalImage,
```
Comment (mathewnik90) [inline goa/Classes/ext/LipiTKNode.cpp:300]: In some places this is pass by reference whereas in others its pass by value

**Review by mathewnik90 — APPROVED** [review]

**Review by mathewnik90 — DISMISSED** [review]

**Review by mathewnik90 — APPROVED** [review]

File: goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/AppActivity.java [hunk 19]
```diff
@@ -80,8 +80,11 @@ of this software and associated documentation files (the "Software"), to deal
     // Return Intent extra
     public static SharedPreferences sharedPref;
     protected static String pathToAppDelegate = null;
+    @SuppressLint("StaticFieldLeak")
```
Comment (mathewnik90) [inline goa/proj.android-studio/goa/src/main/java/org/cocos2dx/cpp/AppActivity.java:83]: Why is this suppressed

**Pull request merged**